src/Common/ThreadPool.cpp

```diff
@@ -216,17 +265,14 @@ ThreadPoolImpl<Thread>::~ThreadPoolImpl()
 template <typename Thread>
 void ThreadPoolImpl<Thread>::finalize()
 {
-    {
-        std::lock_guard lock(mutex);
-        shutdown = true;
-    }
-
-    new_job_or_shutdown.notify_all();
-
-    for (auto & thread : threads)
-        thread.join();
```

Comment:
Here was an access violation because threads remove themselves from `threads`, so at the end of execution `thread.join()` here the object `thread` didn't even exist